### PR TITLE
Derive project.version based on latest tag if unspecified

### PR DESCRIPTION
**Pull request by chavdar**
Add logic to build.gradle set project.version if not already set. Use the latest tag as a version. If the tag starts with "gobblin_", it will be stripped. Thus, the tag 'gobblin_0.5.0' becomes version 0.5.0.

### CODE REVIEW

**Comment by chavdar** [comment]
@ynli @ibuenros  can you review?

**Comment by ibuenros** [comment]
LGTM

File: build.gradle [hunk 1]
```diff
@@ -24,6 +24,29 @@ File getEnvironmentScript()
 
 apply from: environmentScript
 
+if (!project.hasProperty('group') || project.group.length() == 0) {
+    project.group='gobblin'
+}
+
+if (!project.hasProperty('version') || project.version=='unspecified') {
```
Comment (liyinan926) [inline build.gradle]: Spaces around `==`.

File: build.gradle [hunk 2]
```diff
@@ -24,6 +24,29 @@ File getEnvironmentScript()
...
+    def versionOut=new ByteArrayOutputStream()
+    exec {
+        commandLine 'git', 'describe', '--tags'
+        standardOutput versionOut
+    }
+    def tagStr=versionOut.toString().trim()
```
Comment (liyinan926) [inline build.gradle]: Spaces around `=`.

**Comment by liyinan926** [comment]
LGTM other than the minor formatting comments.

File: build.gradle (hunk 1 above)
Comment (chavdar) [inline build.gradle]: Fixed.

File: build.gradle (hunk 2 above)
Comment (chavdar) [inline build.gradle]: Fixed.

**Pull request merged**